kedro-datasets/setup.py

```diff
@@ -6,7 +6,7 @@
 PANDAS = "pandas>=1.3, <3.0"
 SPARK = "pyspark>=2.2, <4.0"
 HDFS = "hdfs>=2.5.8, <3.0"
-S3FS = "s3fs>=0.3.0, <0.5"
+S3FS = "s3fs>=2021.04, <2024.1"  # Upper bound set arbitrarily, to be reassessed in early 2024
```

Comment:
Can we keep the lower bound?

Reply:
Why do you want to keep the lower bound? That's already more than 4years old and incompatible with fsspec's calVer versioning, that we already set in kedro (e.g. for kedro `v0.18.0` https://github.com/kedro-org/kedro/blob/0.18.0/requirements.txt#L6)

Reply:
I agree it should be OK to bump the lower bound, the only thing here is that this PR would then have to wait for the upcoming kedro-datasets 2.0.0 :+1: cc @merelcht

Reply:
@MatthiasRoels You are right, I thought the Kedro pin is at 0.3.0 but it is actually just for tests.
```suggestion
S3FS = "s3fs>=2021.4, <2024.1"  # Upper bound set arbitrarily, to be reassessed in early 2024
```

Not sure if this matter but the version name should be `2021.4.0`